### PR TITLE
KRACOEUS-8494 S2S Print forms: After first print attempt, no message given when printing all forms and there are errors on some forms that prevent them from being printed.

### PR DESCRIPTION
**Pull request by mrudulpolus**
Committed files.

### CODE REVIEW

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerService.java [hunk 1]
```diff
@@ -5,4 +5,6 @@
 public interface S2SErrorHandlerService {
 
     AuditError getError(String key);
+    
+    AuditError getErrors(String key, String formName);
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerService.java]: Please convert all generators to use your new api.  The form name should be available in each usage case.

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerService.java [hunk 2]
```diff
@@ -5,4 +5,6 @@
 public interface S2SErrorHandlerService {
 
     AuditError getError(String key);
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerService.java]: Please remove this old api

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerServiceImpl.java [hunk 3]
```diff
@@ -21,4 +21,10 @@ public AuditError getError(String key) {
         return s2sError == null ? new AuditError(AuditError.NO_FIELD_ERROR_KEY, key + " is not valid", AuditError.GG_LINK) :
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SErrorHandlerServiceImpl.java]: Please remove this old api

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SValidatorService.java [hunk 4]
```diff
@@ -42,5 +42,18 @@
...
+     * 
+     * This method receives an XMLObject and validates it against its schema and returns the validation result. It also receives a
+     * list in which upon validation failure, populates it with XPaths of the error nodes
+     * 
+     * @param formObject XML document as {@link}XMLObject
+     * @param errors List list of XPaths of the error nodes.
+     * @param formName is the FormName.
+     * @return validation result true if valid false otherwise.
+     */
+    boolean validate(XmlObject formObject, List<AuditError> errors, String formName);
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SValidatorService.java]: Could you convert validateUserAttachedFormFile to use this new api.  Then you could remove the old validate method that doesn't take a form name.

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SValidatorServiceImpl.java [hunk 5]
```diff
@@ -67,6 +67,29 @@ public boolean validate(XmlObject formObject, List<AuditError> errors) {
 
         return result;
     }
+
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/validate/S2SValidatorServiceImpl.java]: remove old api if possible.  see above comment.

File: coeus-s2sgen-api/src/main/java/org/kuali/coeus/s2sgen/api/generate/FormGeneratorService.java [hunk 6]
```diff
@@ -58,5 +58,5 @@
      * @throws java.lang.IllegalArgumentException if the s2sUserAttachedFormFile is null
      */
 
-    FormValidationResult validateUserAttachedFormFile(S2sUserAttachedFormFileContract s2sUserAttachedFormFile) throws S2SException;
+    FormValidationResult validateUserAttachedFormFile(S2sUserAttachedFormFileContract s2sUserAttachedFormFile,String formName) throws S2SException;
```
Comment (leoherbie) [inline coeus-s2sgen-api/src/main/java/org/kuali/coeus/s2sgen/api/generate/FormGeneratorService.java]: please add javadoc for formName.  Make sure formName cannot be blank

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/FormGeneratorServiceImpl.java [hunk 7]
```diff
@@ -270,7 +270,7 @@ protected void setFormObject(Forms forms, XmlObject formObject) {
 	}
 
     @Override
-    public FormValidationResult validateUserAttachedFormFile(S2sUserAttachedFormFileContract s2sUserAttachedFormFile) throws S2SException {
+    public FormValidationResult validateUserAttachedFormFile(S2sUserAttachedFormFileContract s2sUserAttachedFormFile,String formName) throws S2SException {
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/FormGeneratorServiceImpl.java]: validate formName cannot be null

**Comment by mrudulpolus** [comment]
Modifications committed.

**Pull request merged**